app/delivery/send_to_providers.py

```diff
@@ -131,10 +131,12 @@ def send_email_to_provider(notification):
                     raise MalwarePendingException
 
             if sending_method == "attach":
+                # Prevent URL patterns like file:// ftp:// that may lead to security vulnerabilities
+                if not personalisation_data[key]["document"]["direct_file_url"].lower().startswith(("http", "s3")):
+                    raise ValueError("Invalid URL provided")
```

Comment:
We should have an except statement for ValueError in `deliver_sms` and `deliver_email` such that sending the notification is not retried. I'm happy to discuss

Reply:
added